src/npcmove.cpp

```diff
@@ -1351,26 +1344,28 @@ void npc::find_item()
         }
     }
 
-    if (fetching_item && is_following()) {
-        say(_("Hold on, I want to pick up that %s."),
-            wanted->tname().c_str());
+    if( fetching_item && is_following() ) {
+        say( _("Hold on, I want to pick up that %s."),
+             wanted->tname().c_str() );
     }
 }
 
 void npc::pick_up_item()
 {
-    add_msg( m_debug, "%s::pick_up_item(); [%d, %d] => [%d, %d]", name.c_str(), posx(), posy(), itx, ity );
-    update_path(itx, ity);
+    add_msg( m_debug, "%s::pick_up_item(); [%d, %d, %d] => [%d, %d, %d]", name.c_str(),
+             posx(), posy(), posz(), wanted_item_pos.x, wanted_item_pos.y, wanted_item_pos.z );
+    update_path( wanted_item_pos );
 
-    if (path.size() > 1) {
-        add_msg( m_debug, "Moving; [%d, %d] => [%d, %d]", posx(), posy(), path[0].x, path[0].y );
+    if( path.size() > 1 ) {
+        add_msg( m_debug, "Moving; [%d, %d] => [%d, %d]", 
```

Comment:
The format string should be changed here like you changed it above to output 2 * 3 integers.